indicator/forum/indicator.class.php

```diff
@@ -89,7 +90,35 @@ protected function get_rawdata($startdate, $enddate) {
         $params['courseid'] = $this->courseid;
         $params['startdate'] = $startdate;
         $params['enddate'] = $enddate;
-        if ($readposts = $DB->get_recordset_sql($sql, $params)) {
+		$readposts = $DB->get_recordset_sql($sql, $params);
+		// check if there is any data in forum_read table
+		if (!$readposts->valid()) {
```

Comment:
It might be better to simply check if the config option that turns forum_read usage on and off is set or not.

If set, it can use the existing query, otherwise it can use the more expensive log scanning.